### DISCUSSION TITLE
Expression-bodied property set notation could be further simplified

### DISCUSSION THREAD

**Discussion by IanKemp** [dotnet/csharplang#8722] [Language Ideas]
The updated expression-bodied property syntax is great, but in my opinion is still a little clumsy regarding setting values. Currently, for a simple read/write property with no other logic, the property looks like this:

```
string label;

public string Label
{
    get => label;
    set => label = value;
}
```

I'm aware this can be accomplished with a standard auto-property, so here's a more realistic example with a delegated property:

```
class ComplexObject
{
    public string AString { get; set; }
}

ComplexObject delegatedTo = new ComplexObject();

public string Label
{
    get => delegatedTo.AString;
    set => delegatedTo.AString = value;
}
```

My issue is with the `= value` part of the statement, which I feel is superfluous because that's what you're going to be writing 99% of the time you're declaring properties in this way. I'd be much happier being able to write:

```
public string Label
{
    get => delegatedTo.AString;
    set => delegatedTo.AString;
}
```

**Comment by alrz** [comment]
With generators you won't even need to type the setter at all,
```cs
[BiDirectional]
public string Label => delegateTo.AString;

// to be generated
public replace string Label
{
  get => delegateTo.AString;
  set => delegateTo.AString = value;
}
```
Or if it's desirable to delegate all members,
```cs
[Mixin]
private readonly ComplexObject delegateTo;
```
And let the generator generate code for member delegation.

**Comment by DavidArno** [comment]
@alrz,

But will generators ever realistically happen?

@IanKemp,

In my view, having an expression after the setter, which is converted to `_field = value` by the compiler saves a tiny bit of typing at the expense of readability.

**Comment by alrz** [comment]
>But will generators ever realistically happen?

Nah.

**Comment by svick** [comment]
How would you deal with the fact that code like the following is already valid?

```c#
string field;

ref string Accessor() => ref field;

public string Label
{
    get => Accessor();
    set => Accessor();
}
```

(Yes, in this specific case, the code has a bug. But there could be cases where similar code is intentional.)

**Comment by jjvanzon** [comment]
Shorter possible notation:
`public string Title { get; set; } => labelTitle.Text;`

**Reply by MaximShey** [reply]
Or even 
`public string Title <=> labelTitle.Text;`
The reason behind this syntax is that "=>" is used to define a read-only property, "<=>" will define a read-write (bidirectional) property.

**Comment by bondsbw** [comment]
If the nested class can be private:
```C#
class ComplexObject
{
    public string AString;
}

ComplexObject delegatedTo = new ComplexObject();
public ref string Label => ref delegatedTo.AString;
```
